### PR TITLE
fix docs for mmdetection 3.x

### PR DESCRIPTION
**Pull request by XuDewen**
# Changes : 
- Fix part of baseline section for the changes of MMDetection 3.x
  - Remove the tutorial of evaluation section and training section from README.md
  - Add a `usage_guide/usage_guide_mmdetection_v3.md` to introduce how to evaluate pretrained models and train a model on LaboroTomato dataset by MMDetection 3.x
  - Add a `usage_guide/usage_guide_mmdetection_v2.md` for MMDetection 2.x old users.
  - A `config_example` has been added as a reference for usage guide.

### CODE REVIEW

File: USAGE_GUIDE_3x.md [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+                     pre_trained_model/laboro_tomato_48ep.pth --show
+
+# multi-gpu testing with 4 GPUs
+./tools/dist_test.sh configs/mask_rcnn/laboro_tomato_mask-rcnn_r50_fpn_1x_coco.py \
+                     pre_trained_model/laboro_tomato_48ep.pth 4 --out results.pkl --eval bbox segm                     
+```
+
+## Train a model
+
+To train your model finish all steps from _Test a model_ section and change learning rate and total epoch, steps at `configs/_base_/schedules/schedule_1x.py`. The default learning rate in config files is for 8 GPUs and 2 img/gpu (`base_batch_size` = 16 (8x2)). According to the [Linear Scaling Rule](https://arxiv.org/abs/1706.02677), you need to set the learning rate proportional to the batch size if you use different GPUs or images per GPU, e.g., lr=0.01 for 4 GPUs x 2 img/gpu and lr=0.08 for 16 GPUs x 4 img/gpu.
```
Comment (k-nai) [inline USAGE_GUIDE_3x.md]: `Test a model section`とは、本ドキュメントの`Evaluation`のことを指すのでしょうか？
その場合、ドキュメント内で使われている表現を用いて説明するようにしてください。
そうでない場合は、`Test a model section`が指す箇所を明示してください。

File: README.md [hunk 2]
```diff
@@ -2,27 +2,24 @@
...
+      - [Output examples](#output-examples)
+  - [Usage Guide](#usage-guide)
+    - [MMDetection v3.x](#mmdetection-v3x)
+    - [MMDetection v2.x](#mmdetection-v2x)
+  - [Subsets](#subsets)
     - [Details](#details)
     - [Pretrained models](#pretrained-models)
-    - [Download subsets](#download-subsets)
-
+    - [Download subests](#download-subests)
```
Comment (k-nai) [inline README.md]: "subests"ではなく、"subsets"に修正してください。

File: README.md [hunk 3]
```diff
@@ -31,15 +28,16 @@
...
+Samples of raw/annotated images: IMG_1066, IMG_1246
 
 ### Annotation details
 
 Each tomato is divided into 2 categories according to size (normal size and cherry tomato) and 3 categories depending on the stage of ripening:  
-* fully_ripened - complitely red color and ready to be harvested. Filled with red color on 90%* or more
-* half_ripened - greenish and needs time to ripen. Filled with red color on 30-89%*
-* green - complitely green/white, sometimes with rare red parts. Filled with red color on 0-30%*
+
+- fully_ripened - complitely red color and ready to be harvested. Filled with red color on 90%* or more
```
Comment (k-nai) [inline README.md]: "complitely"ではなく、"completely"に修正してください。

File: README.md [hunk 4]
```diff
@@ -31,15 +28,16 @@
...
+- half_ripened - greenish and needs time to ripen. Filled with red color on 30-89%*
+- green - complitely green/white, sometimes with rare red parts. Filled with red color on 0-30%*
```
Comment (k-nai) [inline README.md]: "complitely"ではなく、"completely"に修正してください。

File: README.md [hunk 5]
```diff
@@ -68,175 +66,59 @@ image_resolutions: 3024x4032, 3120x4160
...
-* Harvesting forecast based on tomato maturity
-* Automatic harvest of only ripened tomates
-* Identification and automatic thinning of deteriorated and obsolete tomatoes 
-* Sprayig pesticides only on tomatoes at a specific ripening stage 
-* Temperature control in greenhouse according to ripening stage
-* Quality control on production line of food manufactures, etc.
 
-### Licence 
+- Harvesting forecast based on tomato maturity
+- Automatic harvest of only ripened tomates
```
Comment (k-nai) [inline README.md]: "tomates"ではなく"tomatoes"ではないでしょうか？
"tomates"はフランス語のトマトです。

File: README.md [hunk 6]
```diff
@@ -68,175 +66,59 @@ image_resolutions: 3024x4032, 3120x4160
...
+- Identification and automatic thinning of deteriorated and obsolete tomatoes
+- Sprayig pesticides only on tomatoes at a specific ripening stage
```
Comment (k-nai) [inline README.md]: "Sprayig"ではなく、"Spraying"に修正してください。

File: README.md [hunk 7]
```diff
@@ -2,27 +2,24 @@
...
 <!-- TOC -->
 
-- [Overview](#overview)
+- [Laboro Tomato: Instance segmentation dataset](#laboro-tomato-instance-segmentation-dataset)
+  - [Overview](#overview)
     - [About Laboro Tomato](#about-laboro-tomato)
     - [Annotation details](#annotation-details)
     - [Dataset details](#dataset-details)
     - [Scope of application](#scope-of-application)
     - [Licence](#licence)
```
Comment (k-nai) [inline README.md]: "licence"ではなく"license"に修正してください。

File: README.md [hunk 8]
```diff
@@ -68,175 +66,59 @@ image_resolutions: 3024x4032, 3120x4160
...
 
-### Licence 
+- Harvesting forecast based on tomato maturity
+- Automatic harvest of only ripened tomates
+- Identification and automatic thinning of deteriorated and obsolete tomatoes
+- Sprayig pesticides only on tomatoes at a specific ripening stage
+- Temperature control in greenhouse according to ripening stage
+- Quality control on production line of food manufactures, etc.
+
+### Licence
```
Comment (k-nai) [inline README.md]: "licence"ではなく"license"に修正してください。

File: README.md [hunk 9]
```diff
@@ -280,5 +163,5 @@ total epoch = 48
 
 ### Download subests
```
Comment (k-nai) [inline README.md]: "subests"ではなく"subsets"に修正してください。

File: README.md [hunk 10]
```diff
@@ -68,175 +66,59 @@ image_resolutions: 3024x4032, 3120x4160
...
-# single-gpu train
-python tools/train.py configs/mask_rcnn/mask_rcnn_r50_fpn_1x_coco.py \
-                      --work-dir ./laboro_tomato
-
-# multi-gpu train with 4 GPUs
-./tools/dist_test.sh configs/mask_rcnn/mask_rcnn_r50_fpn_1x_coco.py 4 \
-                      --work-dir ./laboro_tomato                    
-```
+### MMDetection v2.x
+MMDetection version 2.x user can refer to [user guide MMDetection 2.x](usage_guide/usage_guide_mmdetection_v2.md).
```
Comment (k-nai) [inline README.md]: リンク先の手順はv2.0、v2.1のバージョンの説明です。
v2.xの表記でいいですが、v2.0とv2.1用の手順であることを明記してください。

File: usage_guide/usage_guide_mmdetection_v3.md [hunk 11]
```diff
@@ -0,0 +1,209 @@
...
+
+It is recommended to symlink the dataset root to $MMDETECTION/data. If your folder structure is different, you may need to change the corresponding paths in config files.
+
+``` text
+mmdetection
+├── mmdet
+├── tools
+├── configs
+├── data
+│   ├── laboro_tomatodat
```
Comment (k-nai) [inline usage_guide/usage_guide_mmdetection_v3.md]: "laboro_tomatodat"ではなく"laboro_tomato"に修正してください。

File: usage_guide/usage_guide_mmdetection_v3.md [hunk 12]
```diff
@@ -0,0 +1,209 @@
...
+                     pre_trained_model/laboro_tomato_48ep.pth --show
+
+# multi-gpu testing with 4 GPUs
+./tools/dist_test.sh configs/mask_rcnn/laboro_tomato_mask-rcnn_r50_fpn_1x_coco.py \
+                     pre_trained_model/laboro_tomato_48ep.pth 4 --out results.pkl --eval bbox segm                     
+```
+
+## 6. Run model training
+
+To train your model finish all steps from _Test a model_ section and change learning rate and total epoch, steps at `configs/_base_/schedules/schedule_1x.py`. The default learning rate in config files is for 8 GPUs and 2 img/gpu (`base_batch_size` = 16 (8x2)). According to the [Linear Scaling Rule](https://arxiv.org/abs/1706.02677), you need to set the learning rate proportional to the batch size if you use different GPUs or images per GPU, e.g., lr=0.01 for 4 GPUs x 2 img/gpu and lr=0.08 for 16 GPUs x 4 img/gpu.
```
Comment (k-nai) [inline usage_guide/usage_guide_mmdetection_v3.md]: "Test a model"がイタリック体になっているので、修正してください。

"Test a model section"とありますが、本ドキュメントに"Test  a model Section"はありません。
ドキュメント内で使われている表現を使用して説明するようにしてください。

File: usage_guide/usage_guide_mmdetection_v3.md [hunk 13]
```diff
@@ -0,0 +1,209 @@
...
+``` python
+dataset_type = 'LaboroTomato'
+data_root = 'data/laboro_tomato/'
+...
+train_dataloader = dict(
+    ...
+    dataset=dict(
+        ...
+        ann_file='annotations/train.json',
+        data_prefix=dict(img='train/')
```
Comment (k-nai) [inline usage_guide/usage_guide_mmdetection_v3.md]: 末尾に","を追加してください。

**Review by k-nai — CHANGES_REQUESTED** [review]
@XuDewen 
コードを確認しました。
何点か指摘事項がありますので確認、修正をお願いします。

以下を確認しました。
* usage guideの表記、説明
* 3.xの手順通りに実行してエラーがないか
  * 以下の箇所の末尾に","を追加すれば、学習まで動作することを確認しました。コメントで指摘していますが、修正をお願いします。
    * https://github.com/laboroai/LaboroTomato/blob/3a24871322118a01cf89c7878db5fd32a0823022/usage_guide/usage_guide_mmdetection_v3.md?plain=1#L94
  * config_exampleのファイルを用いて、評価・学習のコマンドを実行してエラーがないか
    * 問題なく実行できました。

## 全体に関する指摘

* リポジトリのルートにあるREADME.mdにコンフリクトが発生していますので、コンフリクトの解消をお願いします。
